datafusion/physical-expr/src/math_expressions.rs

```diff
@@ -149,6 +160,58 @@ pub fn random(args: &[ColumnarValue]) -> Result<ColumnarValue> {
     Ok(ColumnarValue::Array(Arc::new(array)))
 }
 
+/// Round SQL function
+pub fn round(args: &[ArrayRef]) -> Result<ArrayRef> {
+    if args.len() != 1 && args.len() != 2 {
+        return Err(DataFusionError::Internal(
+            "round function requires one or two arguments".to_string(),
+        ));
+    }
+
+    let mut decimal_places =
+        &(Arc::new(Int64Array::from_value(0, args[0].len())) as ArrayRef);
```

Comment:
It might be more efficient to make a ColumnarValue to avoid creating a large empty array. That would complicate the implementation as it would have to handle both the array argument form and the scalar argument form

Reply:
Ah, there is other function that also create an array like that for one arg case, e.g. `log`, so I will change them all in a follow up.

Reply:
I agree it is reasonable to start with a working implementation and optimize as a follow on 👍